### PR TITLE
fix: Ensure node labels toggle inputs from their own instance

### PR DESCRIPTION
**Pull request by stoberov**
## What does it do?

- Comments out a CSS import of a file that hasn't been committed to the repo
- Removes htmlFor as described in #212

### CODE REVIEW

File: docs/src/stories/DefaultValues/index.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react'
 
 import DropdownTreeSelect from '../../../../src'
-import '../../../../dist/styles.css'
+// import '../../../../dist/styles.css'
```
Comment (mrchief) [inline docs/src/stories/DefaultValues/index.js]: Did you run `yarn build`? This file gets created by the build and this import tells Webpack to bundle it for the docs.

**Comment by mrchief** [comment]
Few things before we can get this merged:

 - Need to run tests and update snapshots (you can run `yarn test --update-snapshots`) and commit those files.
 - Need to run `yarn build:docs` to update the bundle for documentation site.
 - Finally, the PR title needs to follow conventional commit message specs.

File: docs/src/stories/DefaultValues/index.js (hunk 1 above)
Comment (stoberov) [reply docs/src/stories/DefaultValues/index.js]: Ow! 
Here's the thing - I cloned the repo, installed the modules and then (as per the README.md) immediately did a `npm run demo`. For some reason, that styles.css was not generated, hence console would throw an invalid import error. 
It makes full sense now & the css file gets created as expected. I've thus reverted the commit which commented out the line.
Thanks.

Comment (mrchief) [reply docs/src/stories/DefaultValues/index.js]: `npm run demo` is for local development. It doesn't use the filesystem, rather uses the in-memory artifacts.

File: __snapshots__/src/tree-node/actions.test.js.md [hunk 2]
```diff
@@ -8,22 +8,20 @@ Generated by [AVA](https://ava.li).
...
 > Snapshot 1
 
-    [
-      <Action
-        actionData={
-          {
-            action: 'action-0',
-            id: undefined,
-          }
+    <Action
```
Comment (mrchief) [inline __snapshots__/src/tree-node/actions.test.js.md]: This file should not have changed. I'm not sure what happened. Are you using React 15 by any chance?

Comment (stoberov) [reply __snapshots__/src/tree-node/actions.test.js.md]: Hey! 
Wow, you're super active, kudos :) 
I myself have been bumping my head against this for the past 30mins. Travis fails due to this test, and yeah - I haven't made anything odd to change the file, other than `yarn test --update-snapshots`.
React is at version @**16.8.3**

File: src/tree-node/node-label.js [hunk 3]
```diff
@@ -49,7 +49,7 @@ class NodeLabel extends PureComponent {
     }
 
     return (
-      <label title={title || label} htmlFor={id}>
+      <label title={title || label}>
         <Checkbox
           name={id}
```
Comment (mrchief) [inline src/tree-node/node-label.js]: Now I'm thinking, the `name={id}` is not unique in the page when you have multiple instances. This is bad HTML semantically. Had these been radio buttons, the problem would have been more severe. 

I added https://github.com/dowjones/react-dropdown-tree-select/issues/215 to keep track of it.

**Comment by mrchief** [comment]
This is looking good. If we can figure out the issue with actions snapshot getting updated, I think it should be good to merge.

File: __snapshots__/src/tree-node/actions.test.js.md (hunk 2 above)
Comment (mrchief) [reply __snapshots__/src/tree-node/actions.test.js.md]: You're using a recent version which also is suspicious 😈 . I'm wondering if the new React is discarding array if it only has one element and treating it as a single dom node (I'm thinking for efficiency). 

To test the theory out, we could add one more item to the array in the [test](https://github.com/dowjones/react-dropdown-tree-select/blob/develop/src/tree-node/actions.test.js#L9) and see if it renders the array back in the snapshot.

Comment (stoberov) [reply __snapshots__/src/tree-node/actions.test.js.md]: Hahaha :) It's never boring with JavaScript's dependencies! :) 
Indeed, I included one more dummy object in the actions array, rebuilt the snapshots and Travis does not complain any more :)

Comment (mrchief) [reply __snapshots__/src/tree-node/actions.test.js.md]: The plot thickens. 🤔 

I wonder if it's React or ava. If it's React, then it means more issues for me down the line. 😕

**Comment by stoberov** [comment]
Hey, super sorry to be a pest! Just checking if there is any chance for a release that maybe includes this PR some time next week? Thanks for a great module!

**Comment by mrchief** [comment]
I was looking into the actions snapshot issue a bit. I tried upgrading to react 16.8.3 and re-ran the tests but found no issues. The snapshot still was the same. See the screenshot below

image

So I'm not sure why you got an updated snapshot. Can you try the following steps:

 - `rm -rf ./node_modules`
 - `yarn install`
 - `yarn test --update-snapshots`
 - commit updated snapshots and push

If all goes as expected, we should have only one updated snapshot ***after*** you push.

Travs build will not fail when you push updated snapshots.

**Comment by ellinge** [comment]
The issue with snapshots is solved in #220 
Bumped enzyme*-versions. Really should use a lock-file to avoid these kind of issues.

Also a bug in with webpack-dev-middleware which needs you to use 3.6.0 if you don't want to get an error when running demo.

**Comment by mrchief** [comment]
> Bumped enzyme*-versions. Really should use a lock-file to avoid these kinds of issues.

We do have `yarn.lock` but I think we need to pin down exact versions instead of caret ranges.

Anyway, thanks for looking into this!

Let's add @ellinge for maintenance /cc @allcontributors[bot]

**Comment by mrchief** [comment]
@stoberov Do you want to go ahead with this or wait for #220 which would take care of the original issue?

**Comment by stoberov** [comment]
> @stoberov Do you want to go ahead with this or wait for #220 which would take care of the original issue?

Hey! Think #220 is the better solution.

**Comment by mrchief** [comment]
Thanks @stoberov! 

Closing in favor of #220

**Pull request closed without merging**